projects/gloo/pkg/plugins/kubernetes/uds_convert.go

```diff
@@ -20,6 +21,37 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// these labels are used to propagate internal data
+// on synthetic Gloo resources generated from other Kubernetes
+// resources (generally Service).
+const (
+	// KubeSourceResourceLabel indicates the kind of resource that the synthetic
+	// resource is based on.
+	KubeSourceResourceLabel = "~internal.solo.io/kubernetes-source-resource"
+	// KubeSourceResourceLabel indicates the original name of the resource that
+	// the synthetic resource is based on.
+	KubeNameLabel = "~internal.solo.io/kubernetes-name"
+	// KubeSourceResourceLabel indicates the original namespace of the resource
+	// that the synthetic resource is based on.
+	KubeNamespaceLabel = "~internal.solo.io/kubernetes-namespace"
+	// KubeSourceResourceLabel indicates the service port when applicable.
+	KubeServicePortLabel = "~internal.solo.io/kubernetes-service-port"
+)
+
+// ClusterNameForKube builds the cluster name based on _internal_ labels.
+// All of the kind, name, namespace and port must be provided.
+func ClusterNameForKube(us *v1.Upstream) (string, bool) {
+	labels := us.Metadata.GetLabels()
```

Comment:
```suggestion
	labels := us.GetMetadata().GetLabels()
```

Our ci will block this via a getterchecker requirement. Ie if there is a getter use that as a nil safe way of chaining. 🤷